test/test_mkldnn.py

```diff
@@ -182,6 +182,15 @@ def test_add(self):
         mx += my
         self.assertEqual(x, mx.to_dense())
 
+    def test_view(self):
+        x = torch.randn(3, 4, 5, dtype=torch.float32) * 10
+        size = (x.size(0), -1)
+
+        self.assertEqual(
+            x.view(size),
```

Comment:
this doesn't actually check that view shares memory.

Reply:
Is memory sharing an API semantic of `view` (and others like `narrow`) or just a zero-copy optimization?

Reply:
So in general, it's an API semantics of .view as people rely on taking view and then applying mutability: https://pytorch.org/docs/stable/tensors.html#torch.Tensor.view

Would it be plausible to NOT implement view and instead implement `reshape` only?

(we should implement `reshape` anyway - it allows to return a copy)